gix-status/src/index_as_worktree/function.rs

```diff
@@ -270,16 +270,12 @@ impl<'index> State<'_, 'index> {
         }
         let path = entry.path_in(self.path_backing);
         let is_excluded = pathspec
-            .pattern_matching_relative_path(
-                path,
-                Some(entry.mode.is_submodule()),
-                &mut |relative_path, case, is_dir, out| {
-                    self.attr_stack
-                        .set_case(case)
-                        .at_entry(relative_path, Some(is_dir_to_mode(is_dir)), objects)
-                        .map_or(false, |platform| platform.matching_attributes(out))
-                },
-            )
+            .pattern_matching_relative_path(path, true, &mut |relative_path, case, is_dir, out| {
```

Comment:
I don't get why this defaults to `true` now, and would hope there is a test to catch it. If not, definitely a reason to add a couple, probably including a baseline to validate Git works similarly.